packages/components/src/mobile/keyboard-aware-flat-list/use-text-input-offset.native.js

```diff
@@ -0,0 +1,52 @@
+/**
+ * WordPress dependencies
+ */
+import RCTAztecView from '@wordpress/react-native-aztec';
+import { useCallback } from '@wordpress/element';
+
+/** @typedef {import('@wordpress/element').RefObject} RefObject */
+/**
+ * Hook that calculates the currently focused TextInput's current
+ * caret Y coordinate position.
+ *
+ * @param {boolean}   scrollEnabled Whether the scroll is enabled or not.
+ * @param {RefObject} scrollViewRef ScrollView reference.
+ * @return {[Function]} Function to get the current TextInput's offset.
+ */
+export default function useTextInputOffset( scrollEnabled, scrollViewRef ) {
+	const getTextInputOffset = useCallback(
+		async ( caret ) => {
+			const { caretY = null } = caret ?? {};
+			const textInput =
+				RCTAztecView.InputState.getCurrentFocusedElement();
+
+			return new Promise( ( resolve ) => {
+				if (
+					scrollViewRef.current &&
+					textInput &&
+					scrollEnabled &&
+					caretY !== null
+				) {
+					textInput.measureLayout(
+						scrollViewRef.current,
+						( _x, y, _width, height ) => {
+							const caretYOffset =
+								// For cases where the caretY value is -1
+								// we use the y + height value.
```

Comment:
What is the case where `caretY` would be `-1`? It may be helpful to state that here in the comment.

Reply:
Sure! I updated the comment in https://github.com/WordPress/gutenberg/pull/48791/commits/20c9ab03deb8e3bd68d37a56f7ec8dcb041dff4e